src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -167,14 +176,28 @@ public sealed class CultureAwareComparer : StringComparer
     {
         private readonly CompareInfo _compareInfo; // Do not rename (binary serialization)
         private readonly bool _ignoreCase; // Do not rename (binary serialization)
+        private CompareOptions _compareOptions;
 
         internal CultureAwareComparer(CultureInfo culture, bool ignoreCase)
         {
             _compareInfo = culture.CompareInfo;
             _ignoreCase = ignoreCase;
+            _compareOptions = _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
         }
 
-        private CompareOptions Options => _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
+        internal CultureAwareComparer(CultureInfo culture, CompareOptions compareOptions)
+        {
+            _compareInfo = culture.CompareInfo;
+            _compareOptions = compareOptions;
+        }
+
+        private CompareOptions Options
```

Comment:
this property isn't needed. Just passt the `_options` field to the `CompareInfo.Compare` calls

Reply:
I agree with Viktor. this property is not needed

---
In reply to: [167567810](https://github.com/dotnet/coreclr/pull/16334#discussion_r167567810) [](ancestors = 167567810)